src/core/batch.rkt

```diff
@@ -145,6 +158,81 @@
       [_ node]))
   (unmunge reg))
 
+(define (batch-restore-index batch)
+  (make-hash (for/list ([node (in-vector (batch-nodes batch))]
+                        [n (in-naturals)])
+               (cons node n))))
+
+(define (egg-nodes->batch egg-nodes id->spec input-batch rename-dict)
```

Comment:
Also one thing about this function: it extract and parses from egg IR to herbie IR on fly, before it used to be: 1) extracting, 2) parsing. Now these steps are combined

Reply:
That's cool.